### PR TITLE
Feat/generate types without docker

### LINKED ISSUE #384: Can't generate the types using the cli if I don't have a local instance running
# Bug report

## Describe the bug

I can't generate typescript types with the cli, if I don't have a local supabase setup

When I run this command, i get the following output

```
> supabase gen types typescript --db-url 'postgres://postgres:[my-db-password]@db.[my-project-id].supabase.co:6543/postgres' > types.ts

Error: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Run the command above without. having a local instance of supabase installed.

## Expected behavior

I want to generate the types based on the remote project I have running

## System information

- OS: macOs
- Version of supabase-cli: 1.0.7

### LINKED ISSUE #508: Gen types for schema other then `public`
# Feature request

## Is your feature request related to a problem? Please describe.

We use schemas heavily to organized our database. Currently it seems the `gen` command only generates types for the `public` schema. It would be great to be able to generate types for other schemas too.

## Describe the solution you'd like

Ability to generate types for other schemas.

### PR DESCRIPTION
**Pull request by soedirgo**
## What kind of change does this PR introduce?

Feature

## What is the current behavior?

- cannot generate types without Docker
- cannot specify schemas to generate types for

## What is the new behavior?

- generating types with `--project-id` and `--db-url` doesn't require Docker
- specify schema by setting `--schema` multiple times

## Additional context

Closes #508, closes #384

### CODE REVIEW

File: internal/gen/types/typescript/typescript.go [hunk 1]
```diff
@@ -14,9 +15,69 @@ import (
...
+		return []string{}
+	}
+
+	// Generating types on `projectId` and `dbUrl` should work without `supabase
+	// init` - i.e. we shouldn't try to load the config for these cases.
+
+	if projectId != "" {
+		resp, err := utils.GetSupabase().GetTypescriptTypesWithResponse(ctx, projectId, func(_ context.Context, req *http.Request) error {
+			q := req.URL.Query()
+			q.Add("included_schemas", strings.Join(coalesce(schemas, []string{"public"}), ","))
```
Comment (sweatybridge) [inline internal/gen/types/typescript/typescript.go:34]: Did we document this param in openapi spec? I would expect the generated request type to contain this field.

File: internal/gen/types/typescript/typescript.go [hunk 2]
```diff
@@ -14,9 +15,69 @@ import (
...
+		})
+		if err != nil {
+			return err
+		}
+
+		if resp.JSON200 == nil {
+			return errors.New("failed to retrieve generated types: " + string(resp.Body))
+		}
+
+		fmt.Print(resp.JSON200.Types)
```
Comment (sweatybridge) [inline internal/gen/types/typescript/typescript.go:46]:
```suggestion
		fmt.Println(resp.JSON200.Types)
```

any reason not to print a line break? same for a few other places

File: internal/gen/types/typescript/typescript.go [hunk 3]
```diff
@@ -14,9 +15,69 @@ import (
...
+	}
+
+	if dbUrl != "" {
+		matches := utils.PostgresUrlPattern.FindStringSubmatch(dbUrl)
+		if len(matches) != 3 {
+			return errors.New("URL is not a valid Supabase connection string.")
+		}
+		escaped := fmt.Sprintf(
+			"postgresql://postgres:%s@%s/postgres",
+			url.QueryEscape(matches[1]),
```
Comment (sweatybridge) [inline internal/gen/types/typescript/typescript.go:57]:
```suggestion
			// TODO: support password strings in pgmeta
			url.QueryEscape(matches[1]),
```

File: internal/gen/types/typescript/typescript.go (hunk 1 above)
Comment (sweatybridge) [reply internal/gen/types/typescript/typescript.go:34]: adding the missing spec here https://github.com/supabase/infrastructure/pull/9531

File: internal/gen/types/typescript/typescript.go (hunk 2 above)
Comment (sweatybridge) [reply internal/gen/types/typescript/typescript.go:46]: seems like json response already has line break

Comment (soedirgo) [reply internal/gen/types/typescript/typescript.go:41]: I just assumed the generated types already has it

**Review by sweatybridge — APPROVED** [review]

File: internal/gen/types/typescript/typescript.go (hunk 3 above)
Comment (soedirgo) [reply internal/gen/types/typescript/typescript.go]: Ah missed that one, will take a look

**Pull request merged**